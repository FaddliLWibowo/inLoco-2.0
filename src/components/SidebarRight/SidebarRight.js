import React from 'react'
import LayerSubtitleSpace from '../LayerSubtitle/LayerSubtitleSpace.js'
import HTML5Backend from 'react-dnd-html5-backend'
import { DragDropContext } from 'react-dnd'

const SidebarRight = ({
    layers,
    showSidebarRight,
    onLayerClick,
    orderByLayerOrder,
    onLayerUp,
    onLayerDown,
    onLayerDrop,
    onSidebarRightHideClick,
    onLayerRemove
}) => {
    if (!orderByLayerOrder) {
        orderByLayerOrder = () => { return layers }
    }

    // define base class of element
    var cssClass = 'sidebar-right allow-transition-sidebar-right'
    // if showSidebarRight is false, add class to hide the element
    if (!showSidebarRight) {
        cssClass += ' hide-sidebar-right'
    }
    return (
        <div className={cssClass}>
            <div className="layer-list">
                <h1 className="layer-list--title">Camadas em exibição</h1>
                <a className="layer-list--close-button fa fa-times" onClick={onSidebarRightHideClick} role="button"></a>
                <div className="layer-item-space-container">
                    {layers ?
                        orderByLayerOrder(layers).reverse().map((layer, index) => {
                            return (
                                <LayerSubtitleSpace
                                    layer={layer}
                                    key={index}
                                    onLayerClick={onLayerClick}
                                    onLayerUp={onLayerUp}
                                    onLayerDown={onLayerDown}
                                    onLayerDrop={onLayerDrop}
                                    onLayerRemove={onLayerRemove}
                                />
                            )
                        })
                    : ''}
                </div>
            </div>
        </div>
    )
}
// need to wrap the top most component of your application
// to make children draggable with DragDropContext (set up React DnD).
export default DragDropContext(HTML5Backend)(SidebarRight)
