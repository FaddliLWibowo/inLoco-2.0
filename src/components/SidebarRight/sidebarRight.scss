.sidebar-right {
    background: $layer-content-bg-color;
    position: fixed;
    top: 0;
    right: 0;
    width: $layer-content-width;
    height: 100%;
    overflow-y: auto;
    z-index: 1001; // over Leaflet controls
}

.layer-list {
    position: relative;

    &--title {
        border-bottom: 1px solid #ced1d6;
        color: $layer-list-area-title-color;
        font-size: 22px;
        font-weight: normal;
        padding: 15px;
    }

    &--close-button {
        border: 3px solid;
        border-radius: 50%;
        color: $layer-list-icon-color;
        cursor: pointer;
        font-size: 12px;
        font-weight: normal;
        height: 20px;
        padding: 1px 0px 0 0;
        position: absolute;
        right: 20px;
        text-align: center;
        top: 17px;
        width: 20px;
    }
}
