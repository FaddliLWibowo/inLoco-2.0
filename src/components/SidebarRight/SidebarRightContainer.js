import React from 'react'
import SidebarRight from './SidebarRight'
import { connect } from 'react-redux'
import { toggleLayerInformation, slideLayerUp, slideLayerDown, dropLayer, hideSidebarRight, toggleLayer, removeAllLayers, openModal } from '../../actions/actions.js'

/**
 * @param {Object[]} layers - this is array of layers.
 *
 * This function filters layers using the selected
 * property
 *
 * @return {Object[]} - returns all layers that are selected
 */
const selectedLayers = (layers) => {
    if (!Array.isArray(layers)) {
        return []
    }
    return layers.filter(layer => layer.selected)
}

const mapStateToProps = (state, ownProps) => {
    return {
        layers: selectedLayers(state.layers),
        showSidebarRight: state.showSidebarRight,
        orderByLayerOrder: ownProps.orderByLayerOrder,
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onLayerClick: (item) => {
            dispatch(toggleLayerInformation(item))
        },
        onLayerUp: (item) => {
            dispatch(slideLayerUp(item))
        },
        onLayerDown: (item) => {
            dispatch(slideLayerDown(item))
        },
        onLayerDrop: (dragged, target) => {
            dispatch(dropLayer(dragged, target))
        },
        onSidebarRightHideClick: () => {
            dispatch(hideSidebarRight())
        },
        onLayerRemove: (item) => {
            dispatch(toggleLayer(item))
        },
        onRemoveAllLayers: (item) => {
            dispatch(removeAllLayers())
        },
        onOpenModal: (item) => {
            dispatch(openModal(item))
        },
    }
}

const SidebarRightContainer = connect(
    mapStateToProps,
    mapDispatchToProps
)(SidebarRight)

export default SidebarRightContainer
