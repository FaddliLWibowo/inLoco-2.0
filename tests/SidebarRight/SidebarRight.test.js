import React from 'react';
import SidebarRight from '../../src/components/SidebarRight/SidebarRight.js';
import renderer from 'react-test-renderer';

it('SidebarRight component renders correctly with no data', () => {
    const component = shallow(<SidebarRight/>);
    expect(component).toMatchSnapshot();
});
