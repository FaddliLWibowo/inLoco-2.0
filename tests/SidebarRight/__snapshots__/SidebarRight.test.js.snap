// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarRight component renders correctly with no data 1`] = `
<div
  className="sidebar-right"
>
  <div
    className="layer-list"
  >
    <h1
      className="layer-list--title"
    >
      Camadas em exibição
    </h1>
    <a
      className="layer-list--close-button fa fa-times"
      role="button"
    />
  </div>
</div>
`;
